vike/client/client-routing-runtime/renderPageClientSide.ts

```diff
@@ -203,9 +207,18 @@ async function renderPageClientSide(renderArgs: RenderArgs): Promise<void> {
       // Render page view
       await renderPageView(pageContext)
     } else {
-      let res: Awaited<ReturnType<typeof getPageContextFromHooks_isNotHydration>>
+      let res: Awaited<ReturnType<typeof getPageContextFromHooks_isNotHydration>> | PrefetchedPageContext
       try {
-        res = await getPageContextFromHooks_isNotHydration(pageContext, false)
+        const prefetchedPageContext = globalObjectForPrefetchedPageContext?.prefetchedPageContext
+        if (
+          prefetchedPageContext?.pageContextFromHooks &&
+          '_pageId' in prefetchedPageContext.pageContextFromHooks &&
+          prefetchedPageContext.pageContextFromHooks._pageId === pageContext._pageId
```

Comment:
Sorry, I'm not sure I understand your point. May I ask for more details?

Reply:
I think we can save multiple `pageContext` objects, one per `pageId`. In other words `globalObject.prefetchedPageContext` would be a `Record<string, PrefetchedPageContext>` where `string` is the `pageId`. The idea is that if the user hovers over multiple links, the `pageContext` fetched for previous links can be used instead of being discarded.

Reply:
I understood, thank you!
> The idea is that if the user hovers over multiple links, the pageContext fetched for previous links can be used instead of being discarded.

Then, I have implemented this with ` { pageId: string; prefetchedPageContext: PrefetchedPageContext }[]` rather than `Record<string, PrefetchedPageContext>`(because I struggled to fix type errors in `getPageContextFromHooks.ts`)

Reply:
:+1: We can polish such small details later.